MiniApp/Classes/Storage/MiniAppStatus.swift

```diff
@@ -44,4 +46,26 @@ class MiniAppStatus {
         }
         return nil
     }
+
+    func setCustomPermissions(forMiniApp id: String, permissionList: [MASDKCustomPermissionModel]) {
+        var permissionsToStore = permissionList
+        if !id.isEmpty {
+            if let storedPermissions = getCustomPermissions(forMiniApp: id) {
+                storedPermissions.forEach {
+                    permissionsToStore.append($0)
+                }
+            }
+            if let data = try? PropertyListEncoder().encode(permissionsToStore) {
+                self.miniAppCustomPermissionsDefaults?.set(data, forKey: id)
+            }
+        }
+    }
+
+    func getCustomPermissions(forMiniApp id: String) -> [MASDKCustomPermissionModel]? {
+        if let data = self.miniAppCustomPermissionsDefaults?.data(forKey: id) {
+            let miniAppInfo = try? PropertyListDecoder().decode([MASDKCustomPermissionModel].self, from: data)
```

Comment:
I think we need to return the full list of supported permissions here. So even if a permission has never been requested, it should be returned here as `DENIED`.

Reply:
In Android, I am using `PERMISSION_NOT_AVAILABLE` as default to return when user never asked for a custom permission.

Reply:
@corycaywood , is that true? I was not aware of that requirement(couldn't find it in JIRA too). I will make the changes accordingly 👍

Reply:
@khairul-alam-licon we would never return `PERMISSION_NOT_AVAILABLE` when `getCustomPermissions` as it contains the only a list of supported permissions and its granted status.

Reply:
Got it. Thanks.